src/test/src/services/acspRegistrationService.test.ts

```diff
@@ -0,0 +1,137 @@
+                .rejects.toBe(undefined);
+        });
+
+        it("Should throw an error when acsp api returns a status greater than 400", async () => {
+            mockPostAcspRegistration.mockResolvedValueOnce({
+                httpStatusCode: StatusCodes.NOT_FOUND
+            });
+
+            await expect(postAcspRegistration(session, TRANSACTION_ID, acsp))
+                .rejects.toEqual({ httpStatusCode: StatusCodes.NOT_FOUND });
+        });
+
+        it("Should throw an error if SERVICE UNAVAILABLE returned from SDK", async () => {
+            const HTTP_STATUS_CODE = StatusCodes.SERVICE_UNAVAILABLE;
+            mockPostAcspRegistration.mockResolvedValueOnce({
+                httpStatusCode: HTTP_STATUS_CODE
+            } as Resource<CompanyProfile>);
+
+            await expect(postAcspRegistration(session, TRANSACTION_ID, acsp))
+                .rejects.toEqual({ httpStatusCode: StatusCodes.SERVICE_UNAVAILABLE });
+        });
+
+        it("Should throw an error when acsp api returns no resource", async () => {
+            mockPostAcspRegistration.mockResolvedValueOnce({
+                httpStatusCode: StatusCodes.CREATED
+            });
+
+            await expect(postAcspRegistration(session, TRANSACTION_ID, acsp))
+                .rejects.toEqual({ httpStatusCode: StatusCodes.CREATED });
+        });
+    });
+
+    describe("getAcspRegistration tests", () => {
+        it("Should return a acspRegistration", async () => {
+            const dummyAcspDto: AcspDto = {
+                id: EMAIL_ID,
+                typeOfBusiness: "LIMITED"
+            };
+
+            mockGetAcspRegistration.mockResolvedValueOnce({
+                httpStatusCode: 200,
+                resource: dummyAcspDto
+            } as Resource<AcspDto>);
+
+            const acspDto: AcspDto = await getAcspRegistration(session, TRANSACTION_ID, EMAIL_ID);
+
+            expect(acspDto).toStrictEqual(dummyAcspDto);
+        });
+
+        it("Should throw an error when no acsp api response", async () => {
+            mockGetAcspRegistration.mockResolvedValueOnce(undefined);
+
+            await expect(getAcspRegistration(session, TRANSACTION_ID, EMAIL_ID)).rejects.toBe(undefined);
+        });
+
+        it("Should throw an error when acsp api returns a status greater than 400", async () => {
+            mockGetAcspRegistration.mockResolvedValueOnce({
+                httpStatusCode: 404
+            });
+
+            await expect(getAcspRegistration(session, TRANSACTION_ID, EMAIL_ID)).rejects.toEqual({ httpStatusCode: StatusCodes.NOT_FOUND });
+        });
+
+        it("Should throw an error when acsp api returns no resource", async () => {
+            mockGetAcspRegistration.mockResolvedValueOnce({
+                httpStatusCode: StatusCodes.OK
+            });
+
+            await expect(getAcspRegistration(session, TRANSACTION_ID, EMAIL_ID)).rejects.toEqual({ httpStatusCode: StatusCodes.OK });
```

Comment:
want to understand that for no resource, status code 200/201 is correct?

Reply:
updated the httpresponse to return INTERNAL_SERVER_ERROR